test/src/unit-capi-string_dims.cc

```diff
@@ -788,6 +790,8 @@ int StringDimsFx::tiledb_array_get_non_empty_domain_from_name_wrapper(
           (tiledb_serialization_type_t)tiledb::sm::SerializationType::CAPNP,
           1) == TILEDB_OK);
 
+  tiledb_buffer_free(&buff);
```

Comment:
with your changes, 
appears 'buff' in its declaration above should be init'd to nullptr, 
if the ctx or array ever fail sanity checks in the tiledb_serialize_array_non_empty_domain_all_dimensions() call, 
buff will be left with uncertain value (uninit'd stack value),
and this will be attempting to free whatever that value happens to be.

Reply:
While it's a best practice to default-initialize primitives, the test will fatal on `REQUIRE` before reaching this line. We don't really care about leaks in the event of test failure.